gems/pending/middleware/events/middleware_event_monitor.rb

```diff
@@ -0,0 +1,31 @@
+class MiddlewareEventMonitor
+  def initialize(ems, start_time)
+    @ems = ems
+    @alerts_client = ems.alerts_connect
+    # TODO: can we store and resume where we left off?  To avoid duplicates, only look for events now or later,
+    # so basically we miss events when we're not running...
```

Comment:
@blomquisg actually Azure has the same issue. 
https://github.com/ManageIQ/manageiq/pull/7439/files#diff-703f0a31b282a0170102b0ff38cc68edR30

We could even miss events, in case the Worker got killed and takes some time to start again.

Is this a no issue? Do we store Events or are they discarded once processed?